src/main/java/uk/ac/manchester/beehive/tornado/plugins/service/RunInspectionAction.java

```diff
@@ -41,7 +41,7 @@ public void actionPerformed(@NotNull AnActionEvent e) {
             new EmptySelectionWarningDialog().show();
         }else {
             ArrayList<PsiMethod> methods = TornadoTWTask.getMethods(data);
-            DynamicInspection.process(e.getProject(), methods);
+            DynamicInspection.process(e.getProject(), methods, TornadoTWTask.getCalledMethods(methods), TornadoTWTask.getFields());
```

Comment:
Would it be working if you just pass the TornadoTWTask as argument and access the `getMethods()`, `getCalledMethods()` and `getFields()` methods internally?